File: app/src/main/java/vn/loitp/app/activity/customviews/bottomnavigationbar/bottombar/BottomBarActivity.java
package vn.loitp.app.activity.customviews.bottomnavigationbar.bottombar;

import android.os.Bundle;
import android.support.v4.content.ContextCompat;
import android.view.View;
import android.widget.TextView;

import com.daimajia.androidanimations.library.Techniques;

import loitp.basemaster.R;
import vn.loitp.core.base.BaseFontActivity;
import vn.loitp.core.utilities.LStoreUtil;
import vn.loitp.core.utilities.LUIUtil;
import vn.loitp.views.LToast;
import vn.loitp.views.bottombar.LBottomBar;

public class BottomBarActivity extends BaseFontActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        TextView tv = (TextView) findViewById(R.id.tv);
        tv.setText(LStoreUtil.readTxtFromRawFolder(activity, R.raw.loitp));
        LBottomBar lBottomBar = (LBottomBar) findViewById(R.id.bottom_bar);
        lBottomBar.setColorIvOn(R.color.Red);
        lBottomBar.setColorIvOff(R.color.Pink);
        lBottomBar.setItem0(R.drawable.baseline_bug_report_black_48, "Bug report");
        lBottomBar.setItem1(R.drawable.baseline_add_black_48, "Add");
        lBottomBar.setItem2(R.drawable.baseline_chat_black_48dp, "Chat");
        lBottomBar.setItem3(R.drawable.baseline_clear_black_48, "Clear");
        lBottomBar.setItem4(R.drawable.baseline_cloud_download_black_48, "Cloud");
        lBottomBar.setItem5(R.drawable.baseline_picture_in_picture_alt_white_48dp, "Picture");
        lBottomBar.setTechniques(Techniques.Bounce);
        lBottomBar.setOnItemClick(new LBottomBar.Callback() {
            @Override
            public void OnClickItem(int position) {
                LToast.show(activity, "Touch " + position);
            }
        });
        findViewById(R.id.bt_blur_view_red).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                lBottomBar.getBlurView().setOverlayColor(ContextCompat.getColor(activity, R.color.RedTrans));
            }
        });
        findViewById(R.id.bt_blur_view_green).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                lBottomBar.getBlurView().setOverlayColor(ContextCompat.getColor(activity, R.color.GreenTrans));
            }
        });
        findViewById(R.id.bt_count_1).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                lBottomBar.setCount(1);
            }
        });
        findViewById(R.id.bt_count_3).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                lBottomBar.setCount(3);
            }
        });
        findViewById(R.id.bt_count_5).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                lBottomBar.setCount(5);
            }
        });
        findViewById(R.id.bt_count_6).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                lBottomBar.setCount(6);
            }
        });
        findViewById(R.id.bt_cl_0).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                lBottomBar.setColorTextView(R.color.Red);
            }
        });
        findViewById(R.id.bt_cl_1).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                lBottomBar.setColorTextView(R.color.Black);
            }
        });
        findViewById(R.id.bt_cl_2).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                lBottomBar.setColorTextView(R.color.Blue);
            }
        });
        LUIUtil.setDelay(5000, new LUIUtil.DelayCallback() {
            @Override
            public void doAfter(int mls) {
                if (lBottomBar != null) {
                    lBottomBar.setPerformItemClick(4);
                }
            }
        });
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_bottom_bar;
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/bottomnavigationbar/bottombar/BottomBarActivity.kt
package vn.loitp.app.activity.customviews.bottomnavigationbar.bottombar

import android.os.Bundle
import androidx.core.content.ContextCompat
import com.annotation.IsFullScreen
import com.annotation.LayoutId
import com.annotation.LogTag
import com.core.base.BaseFontActivity
import com.core.utilities.LStoreUtil
import com.core.utilities.LUIUtil
import com.daimajia.androidanimations.library.Techniques
import com.views.bottombar.LBottomBar
import kotlinx.android.synthetic.main.activity_bottom_bar_blur.*
import vn.loitp.app.R

@LayoutId(R.layout.activity_bottom_bar_blur)
@LogTag("BottomBarActivity")
@IsFullScreen(false)
class BottomBarActivity : BaseFontActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        textView.text = LStoreUtil.readTxtFromRawFolder(context = this, nameOfRawFile = R.raw.loitp)
        with(bottomBar) {
            paddingOnInDp = context.resources.getDimension(R.dimen.padding_10).toInt()
            paddingOffInDp = context.resources.getDimension(R.dimen.padding_15).toInt()
            colorIvOn = R.color.red
            colorIvOff = R.color.pink
            setTextMarginBottom(context.resources.getDimension(R.dimen.margin_5).toInt())
            setItem0(R.drawable.ic_bug_report_black_48dp, "Bug report")
            setItem1(R.drawable.ic_add_black_48dp, "Add")
            setItem2(R.drawable.ic_chat_black_48dp, "Chat")
            setItem3(R.drawable.ic_close_black_48dp, "Clear")
            setItem4(R.drawable.ic_cloud_download_black_48dp, "Cloud")
            setItem5(R.drawable.ic_picture_in_picture_alt_black_48dp, "Picture")
            setTechniques(Techniques.Pulse)
            setOnItemClick(object : LBottomBar.Callback {
                override fun onClickItem(position: Int) {
                    showShort("Touch $position")
                }
            })
        }
        btBlurViewRed.setOnClickListener {
            bottomBar.realtimeBlurView.setOverlayColor(ContextCompat.getColor(this, R.color.red50))
        }
        btBlurViewGreen.setOnClickListener {
            bottomBar.realtimeBlurView.setOverlayColor(ContextCompat.getColor(this, R.color.green33))
        }
        btCount1.setOnClickListener {
            bottomBar.setCount(1)
        }
        btCount3.setOnClickListener {
            bottomBar.setCount(3)
        }
        btCount5.setOnClickListener {
            bottomBar.setCount(5)
        }
        btCount6.setOnClickListener {
            bottomBar.setCount(6)
        }
        btShowText.setOnClickListener {
            bottomBar.isAlwayShowText = true
        }
        btHideText.setOnClickListener {
            bottomBar.isAlwayShowText = false
        }
        LUIUtil.setDelay(5000, Runnable {
            bottomBar?.setPerformItemClick(4)
        })
    }

}
